packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -2094,6 +2094,69 @@ function validateTailOptions(
   }
 }
 
+function validateSuspenseListNestedChild(childSlot: mixed) {
+  if (__DEV__) {
+    let isArray = Array.isArray(childSlot);
+    let isIterable = !isArray && typeof getIteratorFn(childSlot) === 'function';
+    if (isArray || isIterable) {
+      warning(
+        false,
+        'A nested %s was passed to <SuspenseList />. This whole array ' +
+          'will be treated as a single row in the list. If this is what you ' +
+          'want, wrap it in a fragment e.g. <>{array}</>. If you meant to ' +
+          'merge multiple arrays to a single list, try using React.Children.toArray(). ' +
+          'If you just wanted to append a row, try using nested SuspenseLists: ' +
+          '<SuspenseList><SuspenseList>{children}</SuspenseList><ExtraChild />' +
+          '</SuspenseList>',
+        isArray ? 'array' : 'iterable',
+      );
+      return false;
+    }
+  }
+  return true;
+}
+
+function validateSuspenseListChildren(
+  children: mixed,
+  revealOrder: SuspenseListRevealOrder,
+) {
+  if (__DEV__) {
+    if (
+      (revealOrder === 'forwards' || revealOrder === 'backwards') &&
+      (children !== undefined && children !== null && children !== false)
+    ) {
+      if (Array.isArray(children)) {
+        for (let i = 0; i < children.length; i++) {
+          if (!validateSuspenseListNestedChild(children[i])) {
```

Comment:
Should we include index in the message? If you’re accidentally including an array (such as if some helper like `renderFoo` returns it under some conditions) you might find it difficult to figure out which item is causing it.